prov/sm2/src/sm2.h

```diff
@@ -299,4 +299,10 @@ int sm2_query_atomic(struct fid_domain *domain, enum fi_datatype datatype,
 		     enum fi_op op, struct fi_atomic_attr *attr,
 		     uint64_t flags);
 
+inline static void sm2_freestack_push(struct sm2_ep *ep,
+				      struct sm2_xfer_entry *xfer_entry)
+{
+	smr_freestack_push(sm2_freestack(ep->self_region), xfer_entry);
```

Comment:
This seems like a rather silly function to have... it's pretty clear what it's doing and is only called twice. what is the purpose of this?

Reply:
well, when I wrote it there were locks involved and ep->self_region were not present.  It was originally much longer to account for that.  Also although it's only used twice during this commit, it will get used a total of 8 times in later commits.

Finally, when scrolling through code with a bunch of `sm2_*` functions, I find it useful to have push be an `sm2_*` function.